library/src/androidTest/java/com/google/android/material/motion/runtime/SchedulerTest.java

```diff
@@ -0,0 +1,163 @@
+  }
+
+  public void testManualPlanPerformanceSchedulerState() {
+    transaction.addNamedPlan(new ManualPlan("manual"), "plan", textView);
+    scheduler.commitTransaction(transaction);
+
+    assertTrue(scheduler.getState() == Scheduler.ACTIVE);
+  }
+
+  public void testDelegatePlanPerformanceSchedulerState() {
+    transaction.addNamedPlan(new DelegatedPlan("delegated"), "plan", textView);
+    scheduler.commitTransaction(transaction);
+
+    assertTrue(scheduler.getState() == Scheduler.IDLE);
+  }
+
+  public void testAddingSchedulerListeners() {
+    TestSchedulerListener firstListener = new TestSchedulerListener();
+    TestSchedulerListener secondListener = new TestSchedulerListener();
+    scheduler.addStateListener(firstListener);
+    scheduler.addStateListener(secondListener);
+
+    transaction.addNamedPlan(new ManualPlan("manual"), "plan", textView);
+    scheduler.commitTransaction(transaction);
+
+    assertTrue(firstListener.getState() == Scheduler.ACTIVE);
+    assertTrue(secondListener.getState() == Scheduler.ACTIVE);
+  }
+
+  private static class StandardPlan extends Plan {
+
+    private final String text;
+
+    private StandardPlan(String text) {
+      this.text = text;
+    }
+
+    @Override
+    public Class<? extends Performer> getPerformerClass() {
+      return StandardPlanPerformer.class;
+    }
+  }
+
+  private static class ManualPlan extends Plan {
+
+    private final String text;
+
+    private ManualPlan(String text) {
+      this.text = text;
+    }
+
+    @Override
+    public Class<? extends Performer> getPerformerClass() {
+      return ManualPerformer.class;
+    }
+  }
+
+  private static class DelegatedPlan extends Plan {
+
+    private final String text;
+
+    private DelegatedPlan(String text) {
+      this.text = text;
+    }
+
+    @Override
+    public Class<? extends Performer> getPerformerClass() {
+      return DelegatedPerformer.class;
+    }
+  }
+
+  public static class StandardPlanPerformer extends Performer implements Performer.PlanPerformance {
+
+    @Override
+    public void addPlan(Plan plan) {
+      StandardPlan standardPlan = (StandardPlan) plan;
+      TextView target = getTarget();
+      target.setText(target.getText() + " " + standardPlan.text);
+    }
+  }
+
+  public static class ManualPerformer extends Performer implements Performer.ManualPerformance {
+
+    @Override
+    public int update(float deltaTimeMs) {
+      return Scheduler.ACTIVE;
+    }
+  }
+
+  public static class DelegatedPerformer extends Performer implements Performer.DelegatedPerformance {
+
+    private DelegatedPerformanceCallback callback;
+    private DelegatedPerformanceTokenCallback tokenCallback;
+
+    @Override
+    public void setDelegatedPerformanceCallback(DelegatedPerformanceTokenCallback callback) {
+      this.tokenCallback = callback;
+    }
+
+    @Override
+    public void setDelegatedPerformanceCallback(DelegatedPerformanceCallback callback) {
+      this.callback = callback;
```

Comment:
you can just no-op this callback. This method is deprecated and will be removed soon.